driver/src/orderbook/streamed/orderbook.rs

```diff
@@ -209,4 +209,55 @@ mod tests {
         // Cleanup the file created here.
         assert!(fs::remove_file(test_path).is_ok());
     }
+
+    #[test]
+    fn delete_events_starting_at_block() {
+        let mut orderbook = Orderbook::default();
+        for block_number in 0..3 {
+            orderbook.handle_event_data(
+                EventData::Added(Event::Deposit(Deposit::default())),
+                block_number,
+                0,
+                H256::zero(),
+                0,
+            );
+        }
+        assert_eq!(orderbook.events.len(), 3);
+        orderbook.delete_events_starting_at_block(1);
+        assert_eq!(orderbook.events.len(), 1);
+        assert_eq!(orderbook.events.iter().next().unwrap().0.block_number, 0);
+    }
+
+    #[test]
+    fn events_get_sorted() {
+        let mut orderbook = Orderbook::default();
+        // We are going to keep track of the correct order using the deposit's batch id.
+        let mut add_event = |deposit_id, block_number, log_index| {
+            orderbook.handle_event_data(
+                EventData::Added(Event::Deposit(Deposit {
+                    batch_id: deposit_id,
+                    ..Default::default()
+                })),
+                block_number,
+                log_index,
+                H256::zero(),
+                0,
+            );
+        };
+        // Add a couple of events in random order.
+        add_event(2, 0, 2);
+        add_event(4, 1, 1);
+        add_event(0, 0, 0);
+        add_event(5, 1, 2);
+        add_event(3, 1, 0);
+        add_event(1, 0, 1);
```

Comment:
If the input parameters to add event were aligned with the way `EventSortKeys` were ordered, I would think that these values would be much easier to digest. I think this would be as easy as using 

`block_number, log_index, deposit_id` instead of `deposit_id, block_number, log_index`.

Just an idea.

Reply:
Also, it appears that `deposit_id` is only ever being used as `batch_id`. Does this mean that is should really just be `batch_id`?

Reply:
I think the test is using `batch_id` to just hold the expected order of the deposits.